src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -53,17 +58,26 @@ class Lambda extends RequestStreamHandler {
     val input = read[Input](inputString)
     for {
       config <- ConfigSource.default.loadF[IO, Config]()
+      log = logger.info(Map("batchRef" -> input.batchId))(_)
+
       folderItems <- dynamoClient.getItems[ArchiveFolderDynamoTable, PartitionKey](List(PartitionKey(input.id)), config.dynamoTableName)
       folder <- IO.fromOption(folderItems.headOption)(
         new Exception(s"No folder found for ${input.id} and ${input.batchId}")
```

Comment:
Do exceptions get logged in a similar way to the log lines? I see the log lines include persisted vars to make it easier to filter, would the exception also include these? (Like `batchRef` being included in every log line)

Reply:
The exceptions get logged in a similar way in that there's a json context object attached. The problem with adding the batchRef is that we're catching any errors at the end where the batchRef is unavailable. There's no way of knowing anyway how far the code got before it failed so we might not even have the batchRef at that point anyway.

I'm not sure if what I said makes sense but basically, you can't have the batch ref in the exception logs.

Reply:
I've just noticed that I haven't added the error logging to this repo. I've added it in.